### PR TITLE
Disposal bins can now hold a destination tagger, and adds a few throughout maps.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request
image
_Pictured, a disposal bin with a mounted destination tagger._

Disposal bins now have a dedicated mount where they can hold a destination tagger, allowing for crewmates to more easily send items through disposals to other departments. This can be especially useful when on larger maps, or when cooperating on larger maps with remote resources like on viro/xenobio.

All disposal bins can hold a destination tagger, but select ones on maps have one pre-loaded.
Disposal bins can be removed with an empty hand and right click, and re-attached with right click as well.

## Why It's Good For The Game

This helps assist with crew cooperation on maps where remote or otherwise inaccessible departments have to really work to cooperate and send resources, this chips away at the barrier of entry by giving most departments a way to send resources back and forth as necessary through the disposals network. It ALSO allows for additional opportunities for players to send dangerous items like bombs or grenades through disposals using destination taggers indirectly 👍 

## Changelog

:cl:
add: Disposal bins can now hold destination taggers using right click, and can be removed the same way. Some disposal bins should have some pre-loaded as well.
/:cl:

### CODE REVIEW

**Review by Wallemations — APPROVED** [review]

**Comment by san7890** [comment]
image

hm what does this refer to

File: code/modules/recycling/disposal/bin.dm [hunk 1]
```diff
@@ -327,19 +327,68 @@
...
+	mounted_tagger = null
+	update_appearance(UPDATE_OVERLAYS)
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/obj/machinery/disposal/bin/examine(mob/user)
+	. = ..()
+	if(isnull(mounted_tagger))
+		. += span_notice("\The destination tagger mount is empty.")
+	else
+		. += span_notice("\The [mounted_tagger] is hanging on the side. Right Click to remove.")
```
Comment (san7890) [inline code/modules/recycling/disposal/bin.dm:384]: oh this line. interesting

File: code/modules/recycling/disposal/bin.dm [hunk 2]
```diff
@@ -327,19 +327,68 @@
...
+	balloon_alert(user, "tagger pulled")
+	playsound(src, 'sound/machines/click.ogg', 60, TRUE)
+	mounted_tagger = null
+	update_appearance(UPDATE_OVERLAYS)
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/obj/machinery/disposal/bin/examine(mob/user)
+	. = ..()
+	if(isnull(mounted_tagger))
+		. += span_notice("\The destination tagger mount is empty.")
```
Comment (san7890) [inline code/modules/recycling/disposal/bin.dm:382]:
```suggestion
		. += span_notice("The destination tagger mount is empty.")
```

oh this line lol. good hting we're cathcing it now

**Review by san7890 — CHANGES_REQUESTED** [review]

**Comment by Profakos** [comment]
While I am a bit sad that the incoming unsorted mail nook will get less use, this PR vastly impoves the mail system's usage and convenience. 👍

**Comment by Mickyan** [comment]
This is great though I wish it worked more like the wall mounted defibrillator by snapping back in cause I can definitely see people accidentally throwing it into the bin or stealing it, I think the unreliability of being able to find a tagger is one of the reasons the system rarely sees much use

File: code/modules/recycling/disposal/bin.dm [hunk 3]
```diff
@@ -327,19 +327,68 @@
...
+/obj/machinery/disposal/bin/examine(mob/user)
+	. = ..()
+	if(isnull(mounted_tagger))
+		. += span_notice("The destination tagger mount is empty.")
+	else
+		. += span_notice("\The [mounted_tagger] is hanging on the side. Right Click to remove.")
+
+/obj/machinery/disposal/bin/Destroy()
+	if(!isnull(mounted_tagger))
+		mounted_tagger.forceMove(drop_location())
```
Comment (Cyberboss) [inline code/modules/recycling/disposal/bin.dm:388]: I think it should be `forceMove`d out only in `deconstruct()` and `QDEL_NULL`ed in Destroy

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/recycling/disposal/bin.dm [hunk 4]
```diff
@@ -327,19 +327,73 @@
...
+/obj/machinery/disposal/bin/Destroy()
+	if(!isnull(mounted_tagger))
+		QDEL_NULL(mounted_tagger).forceMove(drop_location())
+		mounted_tagger = null
+	return ..()
+
+/obj/machinery/disposal/bin/deconstruct(disassembled)
+	. = ..()
+	if(!isnull(mounted_tagger))
+		mounted_tagger.forceMove(drop_location())
```
Comment (Cyberboss) [inline code/modules/recycling/disposal/bin.dm:395]:
```suggestion
		mounted_tagger.forceMove(drop_location())
		mounted_tagger = null
```

File: code/modules/recycling/disposal/bin.dm [hunk 5]
```diff
@@ -327,19 +327,73 @@
...
+	. = ..()
+	if(isnull(mounted_tagger))
+		. += span_notice("The destination tagger mount is empty.")
+	else
+		. += span_notice("\The [mounted_tagger] is hanging on the side. Right Click to remove.")
+
+/obj/machinery/disposal/bin/Destroy()
+	if(!isnull(mounted_tagger))
+		QDEL_NULL(mounted_tagger).forceMove(drop_location())
+		mounted_tagger = null
```
Comment (Cyberboss) [inline code/modules/recycling/disposal/bin.dm:389]:
```suggestion
		QDEL_NULL(mounted_tagger)
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — APPROVED** [review]

File: code/modules/recycling/disposal/bin.dm [hunk 6]
```diff
@@ -327,19 +327,73 @@
...
+		. += span_notice("The destination tagger mount is empty.")
+	else
+		. += span_notice("\The [mounted_tagger] is hanging on the side. Right Click to remove.")
+
+/obj/machinery/disposal/bin/Destroy()
+	if(!isnull(mounted_tagger))
+		QDEL_NULL(mounted_tagger)
+	return ..()
+
+/obj/machinery/disposal/bin/deconstruct(disassembled)
```
Comment (Cyberboss) [inline code/modules/recycling/disposal/bin.dm:391]: Apparently this is illegal, check how it's supposed to be done. Maybe by some method deconstruct calls

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/recycling/disposal/bin.dm (hunk 6 above)
Comment (ArcaneMusic) [reply code/modules/recycling/disposal/bin.dm:391]: I'll take a look tonight 👌

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**